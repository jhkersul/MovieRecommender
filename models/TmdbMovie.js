import { List, Map } from 'immutable';
import config from '../config';

/**
 * The Image URL for TMDB
 */
const TMDB_IMAGE_URL = config.TMDB_IMAGE_URL;

/**
 * Returns the image url
 * @param {String} imagePath The image path provided by TMDB
 */
const getImageURLForMovie = imagePath => `${TMDB_IMAGE_URL}${imagePath}`;

/**
 * Returns the parsed movie Map
 * @param {Object} movieParams The movie params provided by TMDB
 */
export const parseMovie = movieParams => Map({
  id: movieParams.id,
  title: movieParams.title,
  posterImageUrl: getImageURLForMovie(movieParams.poster_path),
});

/**
 * Returns the Immutable List of movies parsed
 * @param {Object} moviesParams A raw array of movies that will be parsed
 */
export const parseListMovies = (moviesParams) => {
  // Getting an array of movies parsed
  const moviesArray = moviesParams.map(movieParam => parseMovie(movieParam));
  // Transforming the array to a Immutable List
  const moviesList = List(moviesArray);
  return moviesList;
};
