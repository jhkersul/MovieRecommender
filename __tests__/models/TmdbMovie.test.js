import { parseMovie, parseListMovies } from '../../models/TmdbMovie';


test('The parsed movie must be the snapshot', () => {
  const movie = { id: 1, title: 'Test', poster_path: '/123' };

  expect(parseMovie(movie)).toMatchSnapshot();
});


test('should parse the snapshot', () => {
  const moviesParams = [
    { id: 1, title: 'Test', poster_path: '/123' },
    { id: 2, title: 'Test 2', poster_path: '/123' },
  ];
  // Parsing list
  const listMovies = parseListMovies(moviesParams);

  expect.assertions(2);
  expect(listMovies.get(0)).toMatchSnapshot();
  expect(listMovies.get(1)).toMatchSnapshot();
});
