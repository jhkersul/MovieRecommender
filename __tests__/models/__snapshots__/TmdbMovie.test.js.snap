// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The parsed movie must be the snapshot 1`] = `
Immutable.Map {
  "id": 1,
  "title": "Test",
  "posterImageUrl": "https://image.tmdb.org/t/p/w500/123",
}
`;

exports[`should parse the snapshot 1`] = `
Immutable.Map {
  "id": 1,
  "title": "Test",
  "posterImageUrl": "https://image.tmdb.org/t/p/w500/123",
}
`;

exports[`should parse the snapshot 2`] = `
Immutable.Map {
  "id": 2,
  "title": "Test 2",
  "posterImageUrl": "https://image.tmdb.org/t/p/w500/123",
}
`;
